### PR TITLE
[TG-196] [feat] add confirm dialog on details pages

### PR DESCRIPTION
**Pull request by valakar**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/garage-it/SmartHouse-frontend/pull/145" target="_blank">ReviewNinja

File: src/app/scenarios/scenario-details/wizard/edit-scenario-wizard/edit-scenario-wizard.component.test.ts [hunk 1]
```diff
@@ -144,20 +134,46 @@ describe('EditScenarioWizardComponent', () => {
...
+                    sut.delete(scenario);
+                    expect(ScenarioService.delete).toHaveBeenCalledWith(scenario);
+                });
+
+                it('should navigate back if everything fine', () => {
+                    sut.delete(scenario);
+                    expect(sut.back).toHaveBeenCalled();
+                });
+
+                // it('should not go back if request fails', () => {
```
Comment (ashilay) [inline src/app/scenarios/scenario-details/wizard/edit-scenario-wizard/edit-scenario-wizard.component.test.ts]: xit?

File: src/app/scenarios/scenario-details/wizard/edit-scenario-wizard/edit-scenario-wizard.component.test.ts [hunk 2]
```diff
@@ -75,32 +32,65 @@ describe('EditScenarioWizardComponent', () => {
...
+            ActivatedRoute,
+            Router,
+            DialogService,
+            ViewContainerRef,
+            DeviceListService);
+
+        sut.back = jasmine.createSpy('back');
+
+
+        // TestBed.configureTestingModule({
```
Comment (ashilay) [inline src/app/scenarios/scenario-details/wizard/edit-scenario-wizard/edit-scenario-wizard.component.test.ts]: Lets leave comment next time when commenting a lot of code

File: src/app/scenarios/scenario-details/wizard/edit-scenario-wizard/edit-scenario-wizard.component.test.ts (hunk 1 above)
Comment (valakar) [reply src/app/scenarios/scenario-details/wizard/edit-scenario-wizard/edit-scenario-wizard.component.test.ts]: !fixed

File: src/app/scenarios/scenario-details/wizard/edit-scenario-wizard/edit-scenario-wizard.component.test.ts (hunk 2 above)
Comment (valakar) [reply src/app/scenarios/scenario-details/wizard/edit-scenario-wizard/edit-scenario-wizard.component.test.ts]: !fixed was accidentally commited

File: src/app/scenarios/scenario-details/wizard/edit-scenario-wizard/edit-scenario-wizard.component.test.ts [hunk 3]
```diff
@@ -144,20 +108,40 @@ describe('EditScenarioWizardComponent', () => {
...
+                DialogService.confirm.and.returnValue(Observable.of(false));
+                sut.delete(scenario);
+            });
+
+            it('should show confirm dialog', () => {
+                expect(DialogService.confirm).toHaveBeenCalled;
+            });
+
+            it('should have proper view container ref for dialog', () => {
+                expect(DialogService.confirm.calls.mostRecent().args[0]).toEqual(ViewContainerRef);
```
Comment (ashkanhosseini) [inline src/app/scenarios/scenario-details/wizard/edit-scenario-wizard/edit-scenario-wizard.component.test.ts]: expect(DialogService.confirm).toHaveBeenCalledWith ?

Comment (valakar) [reply src/app/scenarios/scenario-details/wizard/edit-scenario-wizard/edit-scenario-wizard.component.test.ts]: !fixed

**Pull request merged**